### PR TITLE
Add io.github.buonhobo.KatharaGUI

### PR DESCRIPTION
**Pull request by BuonHobo**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
  
KatharaGUI is a simple GUI for the Kathara network emulation tool. 
It allows you to start and stop labs and connect to the containers in a convenient way. 
KatharaGUI requires Docker to be installed and correctly configured on your system. 
This app is written in Python and uses GTK 4, libadwaita, and VTE.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.buonhobo.KatharaGUI.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        {
+          "type": "script",
+          "commands": ["bash -c 'python3 /app/bin/KatharaGUI.py'"],
+          "dest-filename": "KatharaGUI.sh"
+        },
+        {
+          "type": "inline",
+          "dest-filename": "io.github.buonhobo.KatharaGUI.desktop",
+          "contents": "[Desktop Entry]\nVersion=1.0\nType=Application\nName=KatharaGUI\nExec=KatharaGUI.sh\nComment=A simple Kathara GUI\nIcon=io.github.buonhobo.KatharaGUI\nTerminal=false\nCategories=Utility;Application;"
+        }
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.json:60]: Have a desktop file in your github repo and use that, it's a universal standard.

File: io.github.buonhobo.KatharaGUI.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/BuonHobo/KatharaGUI.git",
+          "commit": "9867f64a585dddee90686f61017f23e86e3e0437"
+        },
+        {
+          "type": "script",
+          "commands": ["bash -c 'python3 /app/bin/KatharaGUI.py'"],
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.json:53]: No need for bash -c

Comment (bbhtt) [reply io.github.buonhobo.KatharaGUI.json:53]: Add the python shebang to the file, then there is no need for this script as well.

File: io.github.buonhobo.KatharaGUI.metainfo.xml
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.metainfo.xml:1]: metainfo file should be upstream not here.

File: io.github.buonhobo.KatharaGUI.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    <p> It allows you to start and stop labs and connect to the containers in a convenient way.      </p>
+    <p> KatharaGUI requires Docker to be installed and correctly configured on your system.          </p>
+
+    <p> This app is written in Python and uses GTK 4, libadwaita, and VTE.                           </p>
+  </description>
+
+  <screenshots>
+    <screenshot type="default">
+      <caption>KatharaGUI when connected to a machine</caption>
+      <image>https://github.com/BuonHobo/KatharaGUI/blob/main/data/screenshots/LFGFFaB.png?raw=true</image>
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.metainfo.xml:34]: these needs to be direct raw.githubusercontet.com links no html pages. Use a commit or a tag in place of `main` in the link.

Comment (bbhtt) [reply io.github.buonhobo.KatharaGUI.metainfo.xml:34]: same for below.

**Comment by BuonHobo** [comment]
Thank you  very much @bbhtt for your quick and helpful support.
I addressed all of the comments:
- I put a desktop file in my upstream repo
- Added the python shebang to eliminate the need for a wrapper script
- Upstreamed the metainfo file
- Changed the screenshot links back to direct imgur links (which they were in the first place, I had changed them to github links thinking it would be better)

Let me know if it's all fine now

**Comment by bbhtt** [comment]
Looks like you rebased over master branch or merged master branch of this repo. Can you fix it ?

File: io.github.buonhobo.KatharaGUI.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "modules": [
+    "python3-requirements.json",
+    "python3-pyyaml.json",
+    "python3-hatchling.json",
+    "python3-pygments.json",
+    {
+      "name": "vte",
+      "buildsystem": "meson",
+      "config-opts": ["--buildtype=release", "-Dgtk3=false", "-Dgtk4=true"],
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.json:25]:
```suggestion
      "config-opts": [
        "--buildtype=release",
        "-Dgtk3=false",
        "-Dgtk4=true"
      ],
```

**Comment by BuonHobo** [comment]
Oh you are right. Is it alright if I just delete the .github directory?

**Comment by bbhtt** [comment]
the commits are a bit messy still, try checking out your branch and do `git rebase --onto new-pr`

**Comment by BuonHobo** [comment]
I think it's fixed now. Sorry for wasting your time

**Comment by bbhtt** [comment]
You can start the build

**Comment by bbhtt** [comment]
bot, build io.github.buonhobo.KatharaGUI

**Comment by BuonHobo** [comment]
This build seems to work fine on my computer. 
If you want to test it out you have to install docker and add yourself to the docker group. 
In order to start a lab, you need to select a directory with a lab.conf file inside it, this is a suitable option:
https://github.com/KatharaFramework/Kathara-Labs/tree/main/main-labs/labs-integrating-several-technologies/small-internet-with-dns-and-web-server/kathara-lab_small-internet-with-dns-and-web-server

In that repo there are many more labs you can play with. Let me know if you have any issues

File: io.github.buonhobo.KatharaGUI.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  "sdk": "org.gnome.Sdk",
+  "command": "KatharaGUI",
+  "finish-args": [
+    "--device=all",
+    "--device=dri",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--filesystem=/run/docker.sock",
+    "--filesystem=home",
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.json:14]: why do you need home ? GTK4 filechoosers should use the portal by default.

File: io.github.buonhobo.KatharaGUI.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{
+  "id": "io.github.buonhobo.KatharaGUI",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "47",
+  "sdk": "org.gnome.Sdk",
+  "command": "KatharaGUI",
+  "finish-args": [
+    "--device=all",
+    "--device=dri",
```
Comment (bbhtt) [inline io.github.buonhobo.KatharaGUI.json:9]: which one is needed ?

**Comment by BuonHobo** [comment]
I changed the permissions, now I only have:

  "finish-args": [
    "--socket=fallback-x11",
    "--socket=wayland",
    "--device=dri",
    "--filesystem=/run/docker.sock"
  ],

**Comment by bbhtt** [comment]
You have to keep `share=ipc` that one is needed when running on X11 sessions.

**Comment by BuonHobo** [comment]
I added it back

**Comment by bbhtt** [comment]
bot, build io.github.buonhobo.KatharaGUI

**Comment by BuonHobo** [comment]
Would it be possible to rename the app to Katharsis before it's pushed to flathub?

**Comment by bbhtt** [comment]
The ID? It has to match your GitHub repo name, so you have to rename that too.

Changing the `nane` tag is fine.

**Pull request closed without merging**